crates/uv/src/commands/pip/compile.rs

```diff
@@ -547,7 +547,14 @@ fn cmd(
                 return Some(None);
             }
 
-            // Always skip the `--upgrade-package` flag, and mark the next item as skipped
+            // Always skip the `--upgrade-package` flag
+            if arg.starts_with("--upgrade-package=") || arg.starts_with("-P") {
```

Comment:
This differs from `--find-links` above for short option.

https://github.com/skshetry/uv/blob/93563ba1979e35903b235e8bf79cc5bf564f3c13/crates/uv/src/commands/pip/compile.rs#L531

where we do `arg.starts_with("-f=)`, but that won't work for `-f<uri>` option.


I can rollback this if `-Pruff` or fix `--find-links` to also exclude for `-f<uri>`.

Reply:
But this raw argument check is not enough to exclude arguments. Eg:
```console
uv pip compile -vPruff
```

or, 
```console
uv pip compile ... -vv
```

Reply:
Yeah let's catch `-P=` and `-P<package>` and same for find-links.